runtime/src/bank/partitioned_epoch_rewards/calculation.rs

```diff
@@ -861,4 +969,229 @@ mod tests {
             expected_reward_info,
         );
     }
+
+    fn compare_stake_rewards(
+        expected_stake_rewards: &[StakeRewards],
+        received_stake_rewards: &[StakeRewards],
+    ) {
+        for (i, partition) in received_stake_rewards.iter().enumerate() {
+            let expected_partition = &expected_stake_rewards[i];
+            assert_eq!(partition.len(), expected_partition.len());
+            for reward in partition {
+                assert!(expected_partition.iter().any(|x| x == reward));
+            }
+        }
+    }
+
+    #[test]
+    fn test_recalculate_stake_rewards() {
+        let expected_num_delegations = 4;
+        let num_rewards_per_block = 2;
+        // Distribute 4 rewards over 2 blocks
+        let (bank, _, _) = create_reward_bank(expected_num_delegations, num_rewards_per_block);
+        let rewarded_epoch = bank.epoch();
+
+        // Advance to next epoch boundary to update EpochStakes
+        let (bank, bank_forks) = bank.wrap_with_bank_forks_for_tests();
+        let bank = new_bank_from_parent_with_bank_forks(
+            &bank_forks,
+            bank,
+            &Pubkey::default(),
+            SLOTS_PER_EPOCH,
+        );
```

Comment:
nit: it would be nice to put all of this in `create_reward_bank` since it looks like it's always advanced to the next epoch boundary